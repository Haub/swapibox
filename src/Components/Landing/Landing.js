import React, { Component } from 'react';
import './Landing.css';
import { firstFetch } from '../App/fetchCalls';
import PropTypes from 'prop-types';
// import a song here

class Landing extends Component {
  constructor(props) {
    super(props); 
    this.state = {
      scrollText: {}
    };
  }

  componentDidMount = async () => {
    try {
      const url = 'https://swapi.co/api/films';
      const reaction = await firstFetch(url);
      const newVariable = await this.fetchFilm(reaction.results);
      this.setState({
        scrollText: newVariable[Math.floor(Math.random() * reaction.results.length)]
      });
    } catch (error){
        console.log(error.message)
    }
  }

  fetchFilm = (arrayOfFilmResults) => {
    const crawlText = arrayOfFilmResults.map(film => {
      let text = film.opening_crawl;
      let year = film.release_date;
      let title = film.title;
      return Object.assign({},  {text: text}, {year: year}, {title: title});
    });
    return Promise.all(crawlText);
  }

  render() {
    const { year, title, text } = this.state.scrollText;
    return (
      <div onClick={this.props.removeLanding}>
        <div className="fade"></div>
        <article className="star-wars">
          <div className="crawl">
            <div className="title">
              <h1>{title}</h1>
              <p>{year}</p>
            </div>
            <p>{text}</p>
            <div className="orange">NOW</div>
            <h1 className="red">CLICK HERE ON THE SCREEN</h1>
          </div>
        </article>
      </div>
    );
  }
}

Landing.propTypes = {
  title: PropTypes.string,
  year: PropTypes.string,
  text: PropTypes.string,
  removeLanding: PropTypes.func
};

export default Landing;
