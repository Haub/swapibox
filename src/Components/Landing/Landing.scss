.star-wars {
  display: flex;
  justify-content: center;
  height: 800px;
  perspective: 400px;
  color: #feda4a;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 500%;
  font-weight: 600;
  letter-spacing: 6px;
  line-height: 150%;
  text-align: justify;
}

.crawl {
  position: relative;
  top: -100px;
  transform-origin: 50% 100%;
}

@keyframes crawl {
  0% {
    top: 0;
    transform: rotateX(20deg) translateZ(0);
  }
  100% { 
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}

.crawl {
  position: relative;
  transform-origin: 50% 100%;
  animation: crawl 40s linear;
}

.fade {
  position: relative;
  width: 100%;
  min-height: 40vh;
  top: -25px;
  background-image: linear-gradient(0deg, transparent, black 75%);
  z-index: 1;
}

.orange {
  color: orange;
}

.red {
  color: red;
}

.crawl p {
  text-shadow: 3px 3px purple;
}