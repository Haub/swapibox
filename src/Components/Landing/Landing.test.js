import React from 'react';
import Landing from './Landing';
import { shallow } from 'enzyme';

describe('Landing', () => {
  let wrapper;
  let mockScroll;
  let mockState;
  let mockEvent;

  beforeEach(() => {
    wrapper = shallow(<Landing />);
    mockEvent = jest.fn();
  });

  // it('matches the snapshot', () => {
  //   expect(wrapper).toMatchSnapshot();
  // });

  // it('has a property scrollText with initial state of an empty object', () => {
  //   mockState = {};
  //   expect(wrapper.state('scrollText')).toBe(mockState);
  // });

  // it('should invoke the fetchFilm function on pageload', () => {
  //   wrapper.instance().fetchFilm(mockEvent);
  //   expect(mockEvent).toHaveBeenCalled();
  // });

  // it('should call fetchFilm with the correct params', () => { 
  //   let mockArray = [{openingCrawl:'Turmoil has engulfed the Galactic Republic. The taxation of trade routes to outlying star systems is in dispute.'}];
  //   expect(fetchFilm).toHaveBeenCalledWith(mockArray);
  // });

  // it('should update state with the correct data when fetchFilm is invoked', () => {
  //   let mockArray = [{openingCrawl:'Turmoil has engulfed the Galactic Republic. The taxation of trade routes to outlying star systems is in dispute.'}];
  //   wrapper.instance().fetchFilm(mockArray);
  //   expect(wrapper.state('scrollText')).toEqual(mockState);
  // });

});