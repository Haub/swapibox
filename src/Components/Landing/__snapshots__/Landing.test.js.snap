// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Landing />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="fade"
/>,
        <article
          className="star-wars"
>
          <div
                    className="crawl"
          >
                    <div
                              className="title"
                    >
                              <h1 />
                              <p />
                    </div>
                    <p />
                    <div
                              className="orange"
                    >
                              NOW
                    </div>
                    <h1
                              className="red"
                    >
                              CLICK HERE ON THE SCREEN
                    </h1>
          </div>
</article>,
      ],
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "fade",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="crawl"
>
            <div
                        className="title"
            >
                        <h1 />
                        <p />
            </div>
            <p />
            <div
                        className="orange"
            >
                        NOW
            </div>
            <h1
                        className="red"
            >
                        CLICK HERE ON THE SCREEN
            </h1>
</div>,
          "className": "star-wars",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title"
>
                <h1 />
                <p />
</div>,
              <p />,
              <div
                className="orange"
>
                NOW
</div>,
              <h1
                className="red"
>
                CLICK HERE ON THE SCREEN
</h1>,
            ],
            "className": "crawl",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1 />,
                  <p />,
                ],
                "className": "title",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "NOW",
                "className": "orange",
              },
              "ref": null,
              "rendered": "NOW",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "CLICK HERE ON THE SCREEN",
                "className": "red",
              },
              "ref": null,
              "rendered": "CLICK HERE ON THE SCREEN",
              "type": "h1",
            },
          ],
          "type": "div",
        },
        "type": "article",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="fade"
/>,
          <article
            className="star-wars"
>
            <div
                        className="crawl"
            >
                        <div
                                    className="title"
                        >
                                    <h1 />
                                    <p />
                        </div>
                        <p />
                        <div
                                    className="orange"
                        >
                                    NOW
                        </div>
                        <h1
                                    className="red"
                        >
                                    CLICK HERE ON THE SCREEN
                        </h1>
            </div>
</article>,
        ],
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fade",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="crawl"
>
              <div
                            className="title"
              >
                            <h1 />
                            <p />
              </div>
              <p />
              <div
                            className="orange"
              >
                            NOW
              </div>
              <h1
                            className="red"
              >
                            CLICK HERE ON THE SCREEN
              </h1>
</div>,
            "className": "star-wars",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="title"
>
                  <h1 />
                  <p />
</div>,
                <p />,
                <div
                  className="orange"
>
                  NOW
</div>,
                <h1
                  className="red"
>
                  CLICK HERE ON THE SCREEN
</h1>,
              ],
              "className": "crawl",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h1 />,
                    <p />,
                  ],
                  "className": "title",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "h1",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "p",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "NOW",
                  "className": "orange",
                },
                "ref": null,
                "rendered": "NOW",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "CLICK HERE ON THE SCREEN",
                  "className": "red",
                },
                "ref": null,
                "rendered": "CLICK HERE ON THE SCREEN",
                "type": "h1",
              },
            ],
            "type": "div",
          },
          "type": "article",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
